lightning-invoice/src/payment.rs

```diff
@@ -1440,11 +1805,24 @@ mod tests {
 		}
 	}
 
-	impl<S: Score> Router<S> for TestRouter {
-		fn find_route(
-			&self, _payer: &PublicKey, route_params: &RouteParameters, _payment_hash: &PaymentHash,
-			_first_hops: Option<&[&ChannelDetails]>, _scorer: &S
+	impl Router for TestRouter {
+		fn find_route<S: Score>(
+			&self, payer: &PublicKey, route_params: &RouteParameters, _payment_hash: &PaymentHash,
+			_first_hops: Option<&[&ChannelDetails]>, scorer: &S
 		) -> Result<Route, LightningError> {
+			// Simulate calling the Scorer just as you would in find_route
+			let route = Self::route_for_value(route_params.final_value_msat);
+			for path in route.paths {
+				for hop in path {
+					let usage = ChannelUsage {
+						amount_msat: hop.fee_msat,
+						inflight_htlc_msat: hop.fee_msat,
```

Comment:
Should this be 0? The scorer here will be `AccountForInFlightHtlcs` wrapping `TestScorer`, IIUC. Or does this need to be a value for a particular test? Seems like `inflight_usage_trivial_test` fails when it is 0.